### ISSUE TITLE
[BUG] Issue while migrating to Elsa 3.2.0 to Elsa 3.3.0  latest version in the existing system environment if any existing records in the [Elsa].[Bookmarks] table

### ISSUE THREAD

**Issue by jayachandra21** [elsa-workflows/elsa-core#6239]
## Description
Facing an issues while migrating to Elsa 3.2.0 to Elsa 3.3.0 latest version in the production environment if there are any existing records in the **[Elsa].[Bookmarks]** table. 

## Steps to Reproduce

1. **Detailed Steps**:  
**Migrating the Elsa workflow from Elsa 3.2.0 Elsa 3.3.0** 
 - In the Elsa 3.2.0 version database, [Elsa].[Bookmarks] table has records.
 - Migrate Elsa from 3.2.0 to Elsa 3.3.0.
 - Migration is failed due to the records in the  [Elsa].[Bookmarks] and which result in failing of table creation **[Elsa].[BookmarkQueueItems]** 
![image](https://github.com/user-attachments/assets/276cb908-0fe9-40f7-8080-d0c356fda3b1)
2. **Reproduction Rate**: "every time"

## Expected Behavior
The migration of the Elsa version should not interrupt the existing system in the production environment.

## Actual Behavior
While migrating the Elsa version the following exceptions occurs.
`Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid column name 'TenantId'.`
`Microsoft.Data.SqlClient.SqlException (0x80131904): Invalid object name 'Elsa.BookmarkQueueItems'.`

## Environment
- **Elsa Package Version**:  Elsa 3.2.0 migrating to Elsa 3.3.0 RC4
- **Operating System**: Windows
- **Browser and Version**: Chrome and Edge

## Troubleshooting Attempts
By manually deleting the **_[Elsa].[Bookmarks]_** and by restart the IIS. The migration scripts are executed and  **`Elsa.BookmarkQueueItems`** table created

**Comment by jayachandra21** [comment]
Hello @sfmskywalker Thank you sharing the steps and sample solution for reference.
I am able to reproduce the error and my steps as follows:

**Elsa server : Elsa 3.2.3 and Elsa Studio 3.2.2**
1.Used the SQL Server to reproduce this error and configured the following lines in the Program.cs file.
`elsa.UseWorkflowManagement(management => management.UseEntityFrameworkCore(ef => ef.UseSqlServer("Connection string)));
elsa.UseWorkflowRuntime(runtime => runtime.UseEntityFrameworkCore(ef
 => ef.UseSqlServer("Connection string")));`
2. Created a Bookmark using the Elsa 3.2.0+ version and I have a records in the [Bookmarks] table.
![image](https://github.com/user-attachments/assets/66e482d0-9922-4847-8f75-6bb64273f0ee)

Now, Migrated the Elsa Server+ Studio to Elsa .3.3.0 rc4 and while running the application getting the error as below
![image](https://github.com/user-attachments/assets/7f4d7bd4-d802-432e-9bdf-835cfdc994c5)

Attached the sample repro that I have used to reproduce this error:
[Elsa330RC4.zip](https://github.com/user-attachments/files/18252811/Elsa330RC4.zip)

**Comment by sfmskywalker** [comment]
@jayachandra21 Your RC4 app works fine when I run it. Obviously, I don't have your database, so it created a new one. If you have any way of sharing your database as a backup that I can restore, I can try and reproduce it. Or if you have a different set of steps, I'd be happy to try that as well.

**Comment by jayachandra21** [comment]
Hello @sfmskywalker, Please use the SQL server database.
If you have any bookmarks in the sql server which was created in the Elsa 3.2.2 or Elsa 3.2.3 then migrate to elsa 3.3.0 rc4
Then issue issue occurs.

**Comment by jayachandra21** [comment]
Hello @sfmskywalker attached the database backup for reference.
[Elsa2Reprod.zip](https://github.com/user-attachments/files/18270037/Elsa2Reprod.zip)

**Comment by sfmskywalker** [comment]
@jayachandra21 I see. Here's what I tried so far:

1. Created a SQL Azure database (I'm on a macbook, which means I cannot run SQL Server locally, not even from a Docker container).
2. I started your project targeting Elsa 3.2.3 core and 3.2.2 studio.
3. I created and executed a workflow that created a bookmark.
4. I upgraded the application to 3.3.0-rc4 and restarted the application.
5. The application successfully starts, which includes a successful migration of the DB.

In other words, I cannot reproduce the reported symptom when working off a new database that was initialized using Elsa 3.2.3 and then upgrading to 3.3.0-rc4.

I'll take a look at your attached database as a final attempt to reproduce the issue.

**Comment by sfmskywalker** [comment]
@jayachandra21 For some reason, the BAK file fails to be imported into SQL Azure. Perhaps you can try sending a BACPAC instead, or an export in CSV format?

**Comment by jayachandra21** [comment]
> @jayachandra21 For some reason, the BAK file fails to be imported into SQL Azure. Perhaps you can try sending a BACPAC instead, or an export in CSV format?

Please find the attached bacpac format for reference.
[Elsa2Reprod_bacpac.zip](https://github.com/user-attachments/files/18270552/Elsa2Reprod_bacpac.zip)

**Comment by jayachandra21** [comment]
@sfmskywalker Did it help with the BACPAC file?

**Comment by sfmskywalker** [comment]
@jayachandra21 Yes, I was able to import the BACPAC file successfully, thank you.
However, when I upgrade your application from 3.2.3 to 3.3.0 RC4 and then start the application, I don't see any issues. The application starts normally and executes the migrations without a glitch. No FAIL messages as seen in your screenshot.

Unless you have already tried, I suggest that you try your own repro as well. Possibly, it's different from the version that you're actually seeing the issue with?

**Comment by jayachandra21** [comment]
Hello @sfmskywalker Thank you!

I am able to reproduce this error.
Please use the attached database. The difference between the yesterday shared database and this database there should be more records in the Bookmarks table.
![image](https://github.com/user-attachments/assets/d613ac2a-ec68-440d-ab04-1cf3904e14f8)
![image](https://github.com/user-attachments/assets/f96ddbb7-a414-41d1-b50b-b31d8d391b2c)
![image](https://github.com/user-attachments/assets/3c4b4103-e998-4535-a6ab-df3508642f65)

Can you please use the attached database below:
[Elsa2Reprod31.zip](https://github.com/user-attachments/files/18273652/Elsa2Reprod31.zip)

**Comment by sfmskywalker** [comment]
For some reason, SQL Azure fails to import that bacpac file. It doesn't provide any specific details as to why it fails to import. The previous bacpac imported with no issue 🤔

**Comment by jayachandra21** [comment]
> For some reason, SQL Azure fails to import that bacpac file. It doesn't provide any specific details as to why it fails to import. The previous bacpac imported with no issue 🤔

Hello @sfmskywalker,
I have attached the database and also csv format.
csv file format : [data in csv file format.zip](https://github.com/user-attachments/files/18280398/data.in.csv.file.format.zip)
bacpac file format  : 
[Elsa2Reprod31.zip](https://github.com/user-attachments/files/18280406/Elsa2Reprod31.zip)

**Comment by sfmskywalker** [comment]
@jayachandra21 Thank you, I was able to import your bacpac and indeed reproduce the issue. I see what the problem is and will look for a solution. In a nutshell, the issue occurs when having at least 2 Bookmark records. The 3.3 migration adds a new column called "Id" which is set to "" by default, but also must be unique. When having 2 or more records, the default value of "" will of course not be unique, and so the migration will fail to succeed.

**Comment by sfmskywalker** [comment]
This issue is now resolved as of build [2467](https://github.com/elsa-workflows/elsa-core/actions/runs/12559818003).
I have successfully tested it against the provided bacpac.

Thank you for helping me hunt down this issue @jayachandra21 👍🏻
